### ISSUE TITLE
no-types should apply to all contexts

### ISSUE THREAD

**Issue by levithomason** [gajus/eslint-plugin-jsdoc#1446]
## Motivation

I use TypeScript and a factory to create my functions, but the `no-types` rule doesn't work:

```ts
/**
 * @param arg {string}
 */
export const fooFn = factory((arg: string) =>);
```

## Current behavior

It seems `no-types` is ignoring this case because it isn't a `function` or `class` keyword. Even if I set:
```ts
"jsdoc/no-types": ["error", { contexts: ["any"] }],
```

## Desired behavior

Setting `any` contexts, should apply to all contexts, not just a few. I never want types in my JSDoc in any case.

**Comment by brettz9** [comment]
It looks like the issue in this case is that your "type" is actually part of the description. The structure of JSDoc blocks:

```js
/**
 * @param {someOptionalType} someName someDescription
 */
```

So, `no-types` should find `someOptionalType` regardless of context, but if there are curly brackets within the description part, we won't detect that, especially since it may not be intended as a type there given how JSDoc is supposed to be structured.

**Comment by brettz9** [comment]
Can you provide the relevant code as text rather than image?
